subsys/bluetooth/host/classic/l2cap_br.c

```diff
@@ -1469,7 +1469,23 @@ static void l2cap_br_conn_rsp(struct bt_l2cap_br *l2cap, uint8_t ident,
 int bt_l2cap_br_chan_send_cb(struct bt_l2cap_chan *chan, struct net_buf *buf, bt_conn_tx_cb_t cb,
 			     void *user_data)
 {
-	struct bt_l2cap_br_chan *br_chan = BR_CHAN(chan);
+	struct bt_l2cap_br_chan *br_chan;
+
+	if (!buf || !chan) {
```

Comment:
The function bt_l2cap_br_chan_send_cb is used internally now. Whether it needs to do this change?

Reply:
What's the rationale of keeping this internal? Which API should L2CAP-based BR/EDR profiles use for sending data if not this one?

Reply:
`bt_l2cap_chan_send` is used for it.

Reply:
> `bt_l2cap_chan_send` is used for it.

The main benefit you get from `CHECKIF()` is that you get more control through Kconfig of the actual behavior: assert, ignore, or return an error like you're doing now. I'd say this is also useful for internal APIs, but since the existing code base is not particularly consistent on this, I don't think it needs to block this PR.